source/level/LevelGenerator.hx

```diff
@@ -0,0 +1,152 @@
+package level;
+
+import flixel.FlxG;
+
+private class Point {
+	public var x:Int;
+	public var y:Int;
+
+	public function new(x:Int, y:Int) {
+		this.x = x;
+		this.y = y;
+	}
+
+	public function toString():String {
+		return this.x + " " + this.y;
+	}
+}
+
+/**
+ * Contains logic to create random levels.
+ * Call generateLevel() to get a bitmap
+ */
+class LevelGenerator {
+	// Defining static constants
+	private static var MIN_Y_GAP_SIZE(default, never):Int = 3;
+	private static var MIN_X_GAP_SIZE(default, never):Int = 3;
+	private static var MIN_WALL_COUNT(default, never):Int = 5;
+	private static var MAX_WALL_COUNT(default, never):Int = 12;
+	private static var MIN_WALL_SIZE(default, never):Int = 4;
+	private static var MAX_WALL_SIZE(default, never):Int = 12;
+
+	private static var BITMAP_EMPTY_BIT(default, never):Int = 0;
+	private static var BITMAP_WALL_BIT(default, never):Int = 1;
+
+	// Instantiating Instance Variables
+	private var mapWidth:Int;
+	private var mapHeight:Int;
+
+	private var minX:Int;
+	private var minY:Int;
+	private var maxX:Int;
+	private var maxY:Int;
+
+	private var map:Array<Array<Int>>;
+
+	public function new(?mapWidth:Int = 20, ?mapHeight:Int = 20) {
+		this.mapWidth = mapWidth;
+		this.mapHeight = mapHeight;
+	}
+
+	/**
+	 * Creates a randomly generated level in the form of a 2D Array
+	 * @return Array<Array<Int>>	Level map
+	 */
+	public function generateLevel():Array<Array<Int>> {
+		instantiateMap();
+		addBorders();
+		addObstacles();
+
+		return this.map;
+	}
+
+	/**
+	 * Populates this.map with an empty 2D Array with width and height specified previously
+	 */
+	private function instantiateMap() {
+		this.map = [for (x in 0...this.mapWidth) [for (y in 0...this.mapHeight) BITMAP_EMPTY_BIT]];
+	}
+
+	private function addBorders() {
+		// Define bounds for borders
+		this.minX = MIN_X_GAP_SIZE;
+		this.minY = MIN_Y_GAP_SIZE;
+		this.maxX = this.mapWidth - 1 - MIN_X_GAP_SIZE;
+		this.maxY = this.mapHeight - 1 - MIN_Y_GAP_SIZE;
```

Comment:
These variables aren't actually used here, it looks like they're needed in `addObstacles()`. 

I would caution against using instance variables for things that are just needed in one local spot, usually it's cleaner to just make local variables inside the spot they're needed (`addObstacles`, in this case) so you don't risk getting those variables altered again or contaminated in ways you don't expect before they're actually used.